src/main/java/io/anserini/document/TwitterDocument.java

```diff
@@ -0,0 +1,63 @@
+package io.anserini.document;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+
+/**
+ * A Twitter document (status).
+ */
+public class TwitterDocument implements SourceDocument {
+  private static final Logger LOG = LogManager.getLogger(io.anserini.document.twitter.Status.class);
+
+  private String id;
+  private String contents;
+
+  private static final JsonParser JSON_PARSER = new JsonParser();
+
+
+  public SourceDocument readNextRecord(String json) throws IOException {
+    JsonObject obj = null;
+    try {
+      obj = (JsonObject) JSON_PARSER.parse(json);
+    } catch (Exception e) {
+      // Catch any malformed JSON.
+      LOG.error("Error parsing: " + json);
+      return null;
+    }
+
+    if (obj.get("text") == null) {
+      return null;
+    }
+
+    // Do not index retweets
+    try {
+      long retweetStatusID = obj.getAsJsonObject("retweeted_status").get("id").getAsLong();
+      return null;
+    } catch (Exception e) {
+      // Do nothing if the retweet doesn't exist
```

Comment:
This comment is misleading, it threw me. Update it to match what's _actually_ going on. Do nothing if there is no retweeted_status key because it's then not a retweet, it's not that the retweet doesn't exist.